File: 14625.cpp
#include <iostream>
using namespace std;

int main() {
	int start_H, start_M, end_H, end_M;
	cin >> start_H >> start_M >> end_H >> end_M;
	int N; cin >> N;

	int cnt = 0;
	while (1) {
		if (start_H % 10 == N || start_H / 10 == N ||
			start_M % 10 == N || start_M / 10 == N)
			cnt++;

		if (start_H == end_H && start_M == end_M)
			break;

		start_M++;
		if (start_M == 60) {
			start_M = 0;
			start_H++;
		}
	}

	cout << cnt << endl;
}

File: 14625.py
sH, sM = map(int, input().split())
eH, eM = map(int, input().split())
N = int(input())

cnt = 0
while 1:
    if (sH % 10 == N or sH // 10 == N or
        sM % 10 == N or sM // 10 == N):
        cnt += 1

    if sH == eH and sM == eM:
        break

    sM += 1
    if sM == 60:
        sM = 0
        sH += 1

print(cnt)
